File: app/src/main/java/ch/threema/app/activities/ExcludedSyncIdentitiesActivity.java
/*  _____ _
 * |_   _| |_  _ _ ___ ___ _ __  __ _
 *   | | | ' \| '_/ -_) -_) '  \/ _` |_
 *   |_| |_||_|_| \___\___|_|_|_\__,_(_)
 *
 * Threema for Android
 * Copyright (c) 2013-2024 Threema GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

package ch.threema.app.activities;

import ch.threema.app.R;
import ch.threema.app.ThreemaApplication;
import ch.threema.app.services.IdListService;

public class ExcludedSyncIdentitiesActivity  extends IdentityListActivity {
	private IdListService listService;

	@Override
	protected IdListService getIdentityListService() {
		if(this.listService == null) {
			this.listService = ThreemaApplication.getServiceManager().getExcludedSyncIdentitiesService();
		}
		return this.listService;
	}

	@Override
	protected String getBlankListText() {
		return this.getString(R.string.prefs_sum_excluded_sync_identities);
	}
	@Override
	protected String getTitleText() {
		return this.getString(R.string.prefs_title_excluded_sync_identities);
	}

}

File: app/src/main/java/ch/threema/app/activities/ExcludedSyncIdentitiesActivity.kt
package ch.threema.app.activities

import android.content.Context
import ch.threema.android.buildActivityIntent
import ch.threema.app.R
import ch.threema.app.ThreemaApplication
import ch.threema.app.utils.logScreenVisibility
import ch.threema.base.utils.getThreemaLogger
import ch.threema.domain.taskmanager.TriggerSource
import ch.threema.domain.types.IdentityString

private val logger = getThreemaLogger("ExcludedSyncIdentitiesActivity")

class ExcludedSyncIdentitiesActivity : IdentityListActivity() {
    init {
        logScreenVisibility(logger)
    }

    private val identityList: IdentityList? by lazy {
        val excludedSyncIdentitiesService = ThreemaApplication.getServiceManager()?.excludedSyncIdentitiesService
            ?: return@lazy null

        object : IdentityList {
            override fun getAll(): Set<IdentityString> {
                return excludedSyncIdentitiesService.getExcludedIdentities()
            }

            override fun addIdentity(identity: IdentityString) {
                excludedSyncIdentitiesService.excludeFromSync(identity, TriggerSource.LOCAL)
            }

            override fun removeIdentity(identity: IdentityString) {
                excludedSyncIdentitiesService.removeExcludedIdentity(identity, TriggerSource.LOCAL)
            }
        }
    }

    override fun getIdentityListHandle(): IdentityList? {
        return identityList
    }

    override fun getBlankListText(): String {
        return this.getString(R.string.prefs_sum_excluded_sync_identities)
    }

    override fun getTitleText(): String {
        return this.getString(R.string.prefs_title_excluded_sync_identities)
    }

    companion object {
        fun createIntent(context: Context) = buildActivityIntent<ExcludedSyncIdentitiesActivity>(context)
    }
}
